modules/ROOT/pages/visual-debugger-concept.adoc

```diff
@@ -3,19 +3,33 @@ ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 
-Anypoint Studio Visual Debugger allows you to run your application in Debug mode, setting breakpoints to stop execution to check the contents of an event at a previously-specified event processor.
+Anypoint Studio Visual Debugger allows you to run your application from Studio in Debug mode, and to
 
-[NOTE]
-Anypoint Studio Debugger only works with Enterprise Edition runtimes and does not connect to Community Edition runtimes.
+The Visual Debugger works with Enterprise Edition runtimes and does not connect to Community Edition runtimes.
 
-When you run your application in Debug mode, the application stops immediately after executing the event processor with the breakpoint. Using the Mule Debugger View, you can browse through the contents of the message as it exists at that point in the flow, and evaluate a Dataweave expression against the captured event.
+You can configure breakpoints to stop the execution of your application at a specific message processor. Using the Mule Debugger View, you can inspect the content of the message as it exists at that point in the flow, and evaluate a Dataweave expression against the captured event.
 
-The Visual Debugger is completely distinct from the Java Debugger contained in Studio; in fact, you may run both debuggers concurrently on your application.
+When you run your application in debug mode for the first time, Aynpoint Studio prompts you to load the Mule Debugger Perspective.
 
-When debugging on your local machine, Studio Visual Debugger listens for incoming TCP connections on localhost port 6666. This port must be available on your machine. This means not blocked by a firewall or other security software.
+The debugger perspective is composed by a list of preconfigured views that help you debug your application:
+
+* Mule Debugger View
+* Evaluate DataWeave Expressions View.
+* Mule Breakpoints View
+
+== Debug your Project in Studio
+
+When debugging on your local machine, the debugger listens for incoming TCP connections on `localhost` and port `6666`. You can modify this configuration in by clicking *Run* > *Debug configurations* and selecting the *Debug* tab.
+
+To debug your application in Studio:
+
+. Right-click your project in the Package Explorer view.
+. Select *Debug As* > *Mule Application*. +
+Studio asks you for permission to switch to the Debug Perspective.
+. Click *Yes*.
+
+Additionally you can run your application in debug mode by right clicking in the canvas and selecting *Debug Project*, or by clicking the Debug button on your top toolbar image:to-run-debug-mode-e7b7b.png[debug-button]
```

Comment:
Is this a process that you run in addition to using Visual Debugger in Studio, or as an alternative to that?
Why would the audience want to use one over the other, or the two together.

Note that this is a "hidden topic," steps in the narrative, that we would want to make more visible by formatting as a task on its own or maybe as followup steps like...:

4. To additionally run your application in debug mode, either:
    * Right-click...
    * Click the *Debug* button...

Reply:
Copyedit as-is:
```suggestion
Additionally, you can run your application in debug mode by right-clicking in the canvas and selecting *Debug Project*, or by clicking the *Debug* button on your top toolbar image:to-run-debug-mode-e7b7b.png[debug-button].
```